docs/internals-mappings.md

```diff
@@ -0,0 +1,428 @@
+# Internals: Mappings
+
+Each Pandora-generated Terraform Resource is associated with a single version of the API - and the Terraform Schema Model(s) for this Resource are based on the API Version.
+
+Mappings define how to map each Field within each Terraform Schema Model onto each Field within the associated SDK Models (located in `hashicorp/go-azure-sdk`).
+
+> **Note:** Mappings are bidirectional - meaning we're mapping the Terraform Schema Field onto an SDK Field (in Terraform's Create/Update functions) and the SDK Field onto the Terraform Schema Field (in Terraform's Read function).
+
+There are three different types of Mappings:
+
+1. Field Mappings.
+2. Model to Model Mappings.
+3. Resource ID (Segment) Mappings.
+
+In each case, these map the fields in the Terraform Schema onto either the SDK Models - or the Resource ID for a given Resource.
+
+Whilst Mappings define how to map a Type of Field from the Schema to another Field, mapping the value of those fields is left to the Terraform Generator, for example where one field is Required and another is Optional - the Terraform Generator should handle passing either a reference/the value as needed.
```

Comment:
What is meant by `type of Field`. It looks like the types are mentioned down below e.g. `DirectAssignment`, `ModelToModel` etc. but that's not clear here.